src/components/manager.js

```diff
@@ -314,6 +318,27 @@ export class Manager extends Component {
       );
     }
   }
+  _nextUnviewedIndex() {
+    const sortedIndexes = Array.from(this.viewedIndexes).sort((a, b) => a - b);
+    return Math.min(
+      (sortedIndexes[sortedIndexes.length - 1] || 0) + 1,
+      this.state.slideReference.length - 1
+    );
+  }
+  _getOffset(slideIndex) {
+    const { goTo } = this.state.slideReference[slideIndex];
+    const nextUnviewedIndex = this._nextUnviewedIndex();
+    if (goTo && !isNaN(parseInt(goTo))) {
+      const goToIndex = () => {
+        if (this.viewedIndexes.has(goTo - 1)) {
+          return this._nextUnviewedIndex();
+        }
+        return goTo - 1;
+      };
```

Comment:
The `goToIndex()` func prevents the infinite viewing loop. If we’ve already seen that out-of-order slide, let’s go to the next unviewed index.